src/Tizen.Multimedia/AudioManager/AudioDevice.cs

```diff
@@ -104,6 +107,185 @@ public bool IsRunning
+
+            return RetrieveFormats();
+
+            IEnumerable<AudioSampleFormat> RetrieveFormats()
+            {
+                int[] formatsRes = new int[numOfElems];
+
+                Marshal.Copy(formats, formatsRes, 0, (int)numOfElems);
+                Interop.Libc.Free(formats);
+
+                IEnumerable<AudioSampleFormat> res = formatsRes.OfType<AudioSampleFormat>();
+                foreach (AudioSampleFormat f in res)
+                {
+                    Log.Debug(Tag, $"supported sample format:{f}");
+                }
+
+                return res;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the device's sample format.
+        /// </summary>
+        /// <value>The <see cref="AudioSampleFormat"/> of the device.</value>
+        /// <remarks>
+        /// This device should be <see cref="AudioDeviceType.UsbAudio"/> type and <see cref="AudioDeviceIoDirection.Output"/> direction.
+        /// <exception cref="ArgumentException">This device is not valid or is disconnected.</exception>
+        /// <exception cref="InvalidOperationException">This device is not <see cref="AudioDeviceType.UsbAudio"/> type or is not <see cref="AudioDeviceIoDirection.Output"/> direction.</exception>
+        /// </remarks>
+        /// <since_tizen> 5 </since_tizen>
+        public AudioSampleFormat SampleFormat
+        {
+            get
+            {
+                Interop.AudioDevice.GetSampleFormat(_id, out AudioSampleFormat format).
+                    ThrowIfError("Failed to get sample format of the device");
+
+                return format;
+            }
+            set
+            {
+                Interop.AudioDevice.SetSampleFormat(_id, value).
+                    ThrowIfError("Failed to set sample format of the device");
+            }
+        }
+
+        /// <summary>
+        /// Gets the device's supported sample rates.
+        /// </summary>
+        /// <returns>An IEnumerable&lt;uint&gt; that contains supported sample rates.</returns>
+        /// <remarks>
+        /// This device should be <see cref="AudioDeviceType.UsbAudio"/> type and <see cref="AudioDeviceIoDirection.Output"/> direction.
+        /// <exception cref="ArgumentException">This device is not valid or is disconnected.</exception>
+        /// <exception cref="InvalidOperationException">This device is not <see cref="AudioDeviceType.UsbAudio"/> type or is not <see cref="AudioDeviceIoDirection.Output"/> direction.</exception>
+        /// </remarks>
+        /// <since_tizen> 5 </since_tizen>
+        public IEnumerable<uint> GetSupportedSampleRates()
+        {
+            IntPtr rates;
+            uint numOfElems;
+
+            Interop.AudioDevice.GetSupportedSampleRates(_id, out rates, out numOfElems).
+                ThrowIfError("Failed to get supported sample formats");
+
+            return RetrieveRates();
+
+            IEnumerable<uint> RetrieveRates()
```

Comment:
How about extracting mehod to use commonly with GetSupportedSampleFormats()?

Reply:
The latest patch, as a result, implementation of RetrieveXXX() differs from each other because of adding convert rate value in GetSupportedSampleRates().